src/test/controller_mapping_settings_test.cpp

```diff
@@ -0,0 +1,508 @@
+
+#include <gtest/gtest.h>
+
+#include <QDomDocument>
+
+// FIXME (acolombier): here we need the CPP file so the templated methods gets
```

Comment:
Is there a reason that you cannot move the templates to 
legacycontrollermapping.h?

Reply:
There is no reasons, no. I took the decision to keep the implementation in the cpp since this last one [is non-trivial](https://github.com/mixxxdj/mixxx/pull/11300/files#diff-1586fb59c112f1db57a7f61e28367835d0e3d95782cb98552968be94c132daeaR117) IMHO. If you would prefer, I could move it to the header file.

Reply:
Yes, I think that will help to remove some other special treatment.